lib/vssh/libssh.c

```diff
@@ -2629,8 +2629,13 @@ static CURLcode myssh_connect(struct Curl_easy *data, bool *done)
     infof(data, "Known hosts: %s", data->set.str[STRING_SSH_KNOWNHOSTS]);
     rc = ssh_options_set(sshc->ssh_session, SSH_OPTIONS_KNOWNHOSTS,
                          data->set.str[STRING_SSH_KNOWNHOSTS]);
+    if(rc == SSH_OK)
+      /* libssh has two separate options for this. Set both to the same file
+         to avoid surprises */
+      rc = ssh_options_set(sshc->ssh_session, SSH_OPTIONS_GLOBAL_KNOWNHOSTS,
+                           data->set.str[STRING_SSH_KNOWNHOSTS]);
     if(rc != SSH_OK) {
-      failf(data, "Could not set known hosts file path");
+      failf(data, "Could not set the global known hosts file path");
```

Comment:
Perhaps keep separate error messages to avoid saying 'global' when
the non-global setting failed to set? (or make the shared error text
non-specific, though I've no immediate idea what it'd be.)

Reply:
Nah it was a mistake. I meant to keep the original error message. I don't think we need to have separate messages.